bpf/kmesh/probes/probe.h

```diff
@@ -0,0 +1,69 @@
+#ifndef __KMESH_BPF_PROBE_H__
+#define __KMESH_BPF_PROBE_H__
+
+#include "access_log.h"
+#include "metrics.h"
+
+static inline void observe_on_pre_connect(struct bpf_sock *sk)
+{
+    struct sock_storage_data *storage = NULL;
+    if (!sk)
+        return;
+
+    storage = bpf_sk_storage_get(&map_of_sock_storage, sk, 0, BPF_LOCAL_STORAGE_GET_F_CREATE);
+    if (!storage) {
+        BPF_LOG(ERR, PROBE, "bpf_sk_storage_get failed\n");
+        return;
+    }
+
+    storage->connect_ns = bpf_ktime_get_ns();
+    return;
+}
+
+static inline void observe_on_connect(struct bpf_sock *sk, __u8 direction)
```

Comment:
```suggestion
static inline void observe_on_connect_open(struct bpf_sock *sk, __u8 direction)
```

Reply:
`observe_on_connect_established` may be more suitable.